File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <map>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int n = nums.size();
    map<int, int> m;
    
    for(int i=0; i<n; i++)
        m[nums[i]]++;
    
    answer = min((int)m.size(), n/2);
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for(int i : nums){
            map.put(i, map.getOrDefault(i,0)+1);
        }
        
        if(nums.length/2 <= map.size())
            answer = nums.length/2;
        else {
            answer = map.size();
        }
        
        return answer;
    }
}